server/cluster/cluster.go

```diff
@@ -126,7 +82,10 @@ func (s *Server) Create(ctx context.Context, q *cluster.ClusterCreateRequest) (*
 		return nil, err
 	}
 
-	c.ConnectionState = appv1.ConnectionState{Status: appv1.ConnectionStatusSuccessful}
```

Comment:
I've noticed that `Get` method still loads cluster version. This is not required anymore.

https://github.com/argoproj/argo-cd/pull/3604/files#diff-02ca8f1595ce93b75591586acd227047R119